bee-node/bee-node/src/tools/convert.rs

```diff
@@ -0,0 +1,136 @@
+use structopt::StructOpt;
+use thiserror::Error;
+
+#[derive(Clone, Debug, Error)]
+pub enum ConvertError {
+    #[error("invalid Bech32 address length")]
+    InvalidAddressLength(),
+    #[error("Invalid Address")]
+    InvalidAddress(),
+}
+
+#[derive(Clone, Debug, StructOpt)]
+pub enum ConvertTool {
+    /// Converts a Bech32 address to a hex encoded one.
+    Bech32ToHex {
+        #[structopt(long)]
+        bech32: String,
+    },
+    /// Converts a hex encoded address to a Bech32 one.
+    HexToBech32 {
+        #[structopt(long)]
+        hex: String,
+        #[structopt(long)]
+        hrp: String,
+    },
+    /// Converts a hex encoded public key to a Bech32 address.
+    HexPubkeyToBech32 {
+        #[structopt(long)]
+        pubkey: String,
+        #[structopt(long)]
+        hrp: String,
+    },
+}
+
+pub fn exec(tool: &ConvertTool) -> Result<(), ConvertError> {
+    match tool {
+        ConvertTool::Bech32ToHex { bech32 } => {
+            let hex = bech32_to_hex(bech32.as_str());
+            match hex {
+                Ok(_) => println!("Your Hex encoded address is:\t{}", hex.unwrap()),
+                Err(e) => println!("Error: {}", e),
+            }
+        }
+        ConvertTool::HexToBech32 { hex, hrp } => {
+            let bech32 = hex_to_bech32(hex.as_str(), hrp.as_str());
+            match bech32 {
+                Ok(_) => println!("Your Bech32 address is:\t{:?}", bech32.unwrap()),
+                Err(e) => println!("Error: {}", e),
+            }
+        }
+        ConvertTool::HexPubkeyToBech32 { pubkey, hrp } => {
+            let bech32 = hex_public_key_to_bech32_address(pubkey.as_str(), hrp.as_str());
+            match bech32 {
+                Ok(_) => println!("Your Bech32 address is:\t{:?}", bech32.unwrap()),
+                Err(e) => println!("Error: {}", e),
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Transforms bech32 to hex
+fn bech32_to_hex(bech32: &str) -> Result<String, ConvertError> {
+    let address = Address::try_from_bech32(bech32).map_err(|_| ConvertError::InvalidAddress())?;
+    let hex_string = match address {
+        (_, Address::Ed25519(ed)) => ed.to_string(),
+        (_, Address::Alias(alias)) => alias.to_string(),
+        (_, Address::Nft(nft)) => nft.to_string(),
+    };
+    Ok(hex_string)
+}
+
+/// Transforms a hex encoded address to a bech32 encoded address
+fn hex_to_bech32(hex: &str, bech32_hrp: &str) -> Result<String, ConvertError> {
+    let address: Ed25519Address = hex
+        .parse::<Ed25519Address>()
+        .map_err(|_| ConvertError::InvalidAddress())?;
+    Ok(Address::Ed25519(address).to_bech32(bech32_hrp))
+}
+
+/// Transforms a hex encoded public key to a bech32 encoded address
+fn hex_public_key_to_bech32_address(hex: &str, bech32_hrp: &str) -> Result<String, ConvertError> {
+    let mut public_key = [0u8; Ed25519Address::LENGTH];
+    hex::decode_to_slice(&hex, &mut public_key).map_err(|_| ConvertError::InvalidAddressLength())?;
```

Comment:
I think for the public key we should also accept it with `0x` prefix as for the hex encoded address
`let public_key: [u8; Ed25519Address::LENGTH] = prefix_hex::decode(hex)?;` should work